src/sentry/models/exporteddata.py

```diff
@@ -67,7 +75,27 @@ def finalize_upload(self, file, expiration=DEFAULT_EXPIRATION):
         current_time = timezone.now()
         expire_time = current_time + expiration
         self.update(file=file, date_finished=current_time, date_expired=expire_time)
-        # TODO(Leander): Implement email notification
+        self.email_success()
+
+    def email_success(self):
+        from sentry.utils.email import MessageBuilder
+
+        # The following condition should never be true, but it's a safeguard in case someone manually calls this method
+        if self.date_finished is None or self.date_expired is None or self.file is None:
+            # TODO(Leander): Implement logging here
+            return
+        msg = MessageBuilder(
+            subject="Your Download is Ready!",
+            context={
+                "url": absolute_uri(
+                    reverse("sentry-data-export-details", args=[self.organization.slug, self.id])
```

Comment:
What would the access rules be for this endpoint ? Do we expect the user to be logged in to access the page on the device they open the email from.

Reply:
Missed this, but yeah the route is auth protected. As long as they're in the org though they'll have access to the download, but they do need to be signed into sentry